tests/test_picking_batch.py

```diff
@@ -305,3 +307,175 @@ def test14_is_valid_location_dest_failure_unknown_location(self):
         self.assertFalse(
             batch.is_valid_location_dest_id("this location does not exist"),
             "An invalid dest location is wrongly marked as valid")
+
+    def test15_unpickable_item_single_move_line_has_package_success_internal_transfer(self):  # noqa
+        """
+        Tests that the picking is cancelled and an internal transfer is created
+        if a picking type is not specified.
+        """
+        picking, batch = self._create_valid_batch_for_location_tests()
+        picking.update_picking(package_name=self.package_one.name)
+        move_line_id = picking.move_line_ids[0].id
+        reason = 'missing item'
+        batch.unpickable_item(move_line_id=move_line_id,
+                              reason=reason,
+                              picking_type_id=None)
+        picking_type = self.package_one.move_line_ids.picking_id.picking_type_id  # noqa
+
+        self.assertEqual(picking.state, 'cancel')
+        self.assertEqual(batch.state, 'done')
+        self.assertEqual(picking_type.name, 'Internal Transfer')
+
+    def test16_unpickable_item_single_move_line_has_package_success_picking_type_specified(self):  # noqa
+        """
+        Tests that the picking is cancelled and the specified picking type
+        is created for the unpickable stock
+        """
+        picking, batch = self._create_valid_batch_for_location_tests()
+        picking.update_picking(package_name=self.package_one.name)
+        move_line_id = picking.move_line_ids[0].id
+        reason = 'missing item'
+
+        # It doesn't actually matter what the picking type is for this test.
+        # The important thing is that the new picking is of type picking_type
+        picking_type = self.picking_type_pick
+        batch.unpickable_item(move_line_id=move_line_id,
+                              reason=reason,
+                              picking_type_id=picking_type.id)
+        new_picking_type = self.package_one.move_line_ids.picking_id.picking_type_id  # noqa
```

Comment:
one_package.move_line_ids can contain more than one line. Let's find the new picking using MoveLine.search(), and try to find a move line in state not done or cancel with package_id = package_one.

Reply:
MoveLine.search([('package_name', '=', self.package_one.name)])

function call "find_move_lines()" on a package.  will return the move lines that are on ther.  package_one.find_move_lines()